### PR TITLE
feat: store historical karma scores

### PR DESCRIPTION
**Pull request by aahna-ashina**
## Dework Task

Store historical Karma scores (week by week)

## How Has This Been Tested?

- [x] Status checks pass
- [ ] Works on Goerli
- [x] Works on Mainnet
- [x] `npm install; npm run lint; npm run build; npm run start;`

### CODE REVIEW

**Review by AaronAbuUsama — COMMENTED** [review]
the linter is complaining about the unexpected `any` how should we handle this?

**Comment by aahna-ashina** [comment]
@pythonpete32 The status checks seem to be passing?  Are you getting an error locally?

**Review by okhaimie-dev — DISMISSED** [review]
Looks good to me.

**Comment by AaronAbuUsama** [comment]
> @pythonpete32 The status checks seem to be passing? Are you getting an error locally?

nothing major just three linting warnings for use of `any` in `karma/src/generate-datasets.ts`

**Comment by aahna-ashina** [comment]
> three linting warnings for use of `any` in `karma/src/generate-datasets.ts`

@pythonpete32 Fixed:  https://github.com/nation3/nationcred-datasets/pull/8/commits/24d05edec64f097907b9801fc251534501e7f0cf

File: data-sources/karma/src/generate-datasets.ts [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+  let passportId: number
+  for (passportId = 0; passportId < nextId; passportId++) {
+    console.info('passportId:', passportId)
+
+    const signerAddress: string = await getSigner(passportId)
+    console.info('signerAddress:', signerAddress)
+
+    const writer = createObjectCsvWriter({
+      path: `output/karma-${signerAddress}.csv`,
```
Comment (johnmark13) [inline data-sources/karma/src/generate-datasets.ts:33]: What is the logic behind having a file per citizen?

File: data-sources/karma/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    // "noImplicitOverride": true,                       /* Ensure overriding members in derived classes are marked with an override modifier. */
+    // "noPropertyAccessFromIndexSignature": true,       /* Enforces using indexed accessors for keys declared using an indexed type. */
+    // "allowUnusedLabels": true,                        /* Disable error reporting for unused labels. */
+    // "allowUnreachableCode": true,                     /* Disable error reporting for unreachable code. */
+
+    /* Completeness */
+    // "skipDefaultLibCheck": true,                      /* Skip type checking .d.ts files that are included with TypeScript. */
+    "skipLibCheck": true                                 /* Skip type checking all .d.ts files. */
+  }
+}
```
Comment (johnmark13) [inline data-sources/karma/tsconfig.json:11]: Can we get rid of all the commented out rules?

File: data-sources/karma/src/generate-datasets.ts (hunk 1 above)
Comment (aahna-ashina) [reply data-sources/karma/src/generate-datasets.ts:33]: @johnmark13 If we include the ETH address in the filename, it's easier to access the data.  When you press a Citizen profile in the Citizen Directory to view more details, we can easily fetch that citizen's data from their file.

Also file versioning becomes less messy this way.

File: data-sources/karma/tsconfig.json (hunk 2 above)
Comment (aahna-ashina) [reply data-sources/karma/tsconfig.json:11]: @johnmark13 Done:  https://github.com/nation3/nationcred-datasets/pull/8/commits/6b7a795b868b7921b47891ba637369a58d16b1f2

**Pull request merged**